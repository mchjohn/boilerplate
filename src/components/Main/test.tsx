import { render, screen } from '@testing-library/react';

import { Main } from '.';

describe('<Main />', () => {
  it('should render the heading', () => {
    const { container } = render(<Main />);

    expect(
      screen.getByRole('heading', { name: /Advanced React/i })
    ).toBeInTheDocument();

    expect(container.firstChild).toMatchSnapshot();
  });

  it('should render the correctly color', () => {
    const { container } = render(<Main />);

    expect(container.firstChild).toHaveStyle({ 'background-color': '#131316' });
  });
});
