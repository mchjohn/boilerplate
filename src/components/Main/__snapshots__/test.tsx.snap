// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Main /> should render the heading 1`] = `
.c0 {
  width: 100%;
  height: 100%;
  padding: 3rem;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  color: #6a6b70;
  background-color: #131316;
  text-align: center;
}

.c1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  text-align: center;
}

.c2 {
  width: 12rem;
  margin-bottom: 2rem;
}

.c3 {
  font-size: 5rem;
  font-weight: 400;
}

.c4 {
  font-size: 2rem;
  font-weight: 400;
}

.c5 {
  margin-top: 2rem;
  width: min(40rem,100%);
}

<main
  class="c0"
>
  <main
    class="c1"
  >
    <img
      alt="Logo do React, imagem de um Ã¡tomo"
      class="c2"
      src="/img/logo-512.svg"
    />
    <h1
      class="c3"
    >
      Advanced React - Boilerplate
    </h1>
  </main>
  <h2
    class="c4"
  >
    TypeScript, React, NextJs e Styled Components
  </h2>
  <img
    alt="Imagem de um terminal estilo do MacOs"
    class="c5"
    src="/img/icon-512.svg"
  />
</main>
`;
